response_operations_ui/views/reporting_units.py

```diff
@@ -213,12 +213,26 @@ def view_reporting_unit_survey(ru_ref, survey_id):
     if case is not None and iac_controller.is_iac_active(case["iac"]):
         unused_iac = case["iac"]
 
-    logger.info("Successfully gathered data to view reporting unit survey data", ru_ref=ru_ref, survey_id=survey_id)
-
     permissions = {
         "reporting_unit_edit": user_has_permission("reportingunits.edit"),
         "messages_edit": user_has_permission("messages.edit"),
     }
+
+    enrolment_code_hyperlink = ""
+    if permissions["reporting_unit_edit"]:
+        enrolment_code_hyperlink = url_for(
+            "reporting_unit_bp.generate_new_enrolment_code",
+            case_id=case["id"],
+            collection_exercise_id=survey_collection_exercises[0]["id"],
```

Comment:
This doesn't seem to yield a deterministic ordering of the collection exercises. They should be sorted but the `[0]` indexed exercise appears to change per run.

Reply:
Fixed with test data change to `scheduledStartDateTime`